src/main/java/seedu/momentum/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +31,79 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
```

Comment:
The method is > 30 lines. I see you tried to reduce the length of the method with combinePredicates but I think you can abstract more of the logic.